File: 2708/main.go
package main

import (
	"math"
	"fmt"
)

func maxStrength(nums []int) int64 {
	n, z, p := 0, 0, 0
	prod := 1
	mn := math.MinInt32
	for _, num := range nums {
		if num < 0 {
			n++
			prod *= num
			if num > mn {
				mn = num
			}
		} else if num == 0 {
			z++
		} else {
			prod *= num
			p++
		}
	}
	if n == 1 && z == 0 && p == 0 {
		return int64(nums[0])
	}
	if n <= 1 && p == 0 {
		return int64(0)
	}
	if prod < 0 {
		return int64(prod / mn)
	} else {
		return int64(prod)
	}
}

func main() {
	fmt.Println(maxStrength([]int{-4, -5, -4})) // 20
}

File: 2708/main.rs
struct Solution {}

impl Solution {
    pub fn max_strength(nums: Vec<i32>) -> i64 {
        let (mut n, mut z, mut p) = (0, 0, 0);
        let mut prod = 1;
        let mut mn = -100000000;
        for &num in nums.iter() {
            if num < 0 {
                n += 1;
                prod *= num;
                if num > mn {
                    mn = num;
                }
            } else if num == 0 {
                z += 1;
            } else {
                prod *= num;
                p += 1;
            }
        }
        if n == 1 && z == 0 && p == 0 {
            return nums[0] as i64
        }
        if n <= 1 && p == 0 {
            return 0 as i64
        }
        if prod < 0 {
            return (prod / mn) as i64
        } else {
            return prod as i64
        }
    }
}

fn main() {
    println!("{}", Solution::max_strength(vec![-4,-5,-4])) // 20
}